File: isermon-ios/isermon/Sermon.swift
//
//  Sermon.swift
//  isermon
//
//  Created by jim on 12/12/2017.
//  Copyright © 2017年 jimgong. All rights reserved.
//

import Foundation

class Sermon {
    var id: String?
    var title: String
    var description: String?
    var speaker: String?
    var scripture: String?
    var info: String?
    var lang: String?
    var urlLocal: String?
    var urlSource: String?
    var username: String?
    var date: String?
    
    var num_listen: Int?
    var num_like: Int?
    var num_bookmark: Int?

    init?(title: String){
        self.title = title
    }
    
    init?(json: [String: Any]){
        self.id = json["_id"] as? String
        self.title = json["title"] as! String
        self.description = json["description"] as? String
        self.speaker = json["speaker"] as? String
        self.scripture = json["scripture"] as? String
        self.info = json["info"] as? String
        self.lang = json["lang"] as? String
        self.urlLocal = json["urlLocal"] as? String
        self.urlSource = json["urlSource"] as? String
        self.username = json["username"] as? String
        self.date = json["date"] as? String

        self.num_listen = json["num_listen"] as? Int
        self.num_like = json["num_like"] as? Int
        self.num_bookmark = json["num_bookmark"] as? Int
    }
}

File: isermon-android/app/src/main/java/com/jimgong/isermon/Sermon.java
/*
 * Copyright (c) 2016 Razeware LLC
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package com.jimgong.isermon;

import android.content.Context;
import android.util.Log;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.InputStream;
import java.util.ArrayList;

public class Sermon {

    String id;
    String title;
    String description;
    String urlLocal;
    String urlSource;
    String username;
    String date;

    static ArrayList<Sermon> getFromResponse(String response){
        ArrayList<Sermon> sermonList = new ArrayList<>();

        try {
            JSONArray sermons = new JSONArray(response);

            // Get Sermon objects from data
            for(int i = 0; i < sermons.length(); i++){
                JSONObject o = sermons.getJSONObject(i);
                Sermon sermon = new Sermon();

                sermon.id = o.getString("_id");
                sermon.title = o.getString("title");
                if (!o.isNull("description")) {
                    sermon.description = o.getString("description");
                }
                if (!o.isNull("urlLocal")) {
                    sermon.urlLocal = o.getString("urlLocal");
                }
                if (!o.isNull("urlSource")) {
                    sermon.urlSource = o.getString("urlSource");
                }
                if (!o.isNull("username")) {
                    sermon.username = o.getString("username");
                }
                if (!o.isNull("date")) {
                    sermon.date = o.getString("date");
                }

                sermonList.add(sermon);
//                Log.d("Sermon", "found sermon: " + sermon.title);
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }

        return sermonList;
    }

    //test
    public static ArrayList<Sermon> getRecipesFromFile(String filename){
        final ArrayList<Sermon> sermonList = new ArrayList<>();

        try {
            // Load data
            String jsonString = loadJsonFromAsset(filename);
            JSONObject json = new JSONObject(jsonString);
            JSONArray recipes = json.getJSONArray("recipes");

            // Get Sermon objects from data
            for(int i = 0; i < recipes.length(); i++){
                Sermon sermon = new Sermon();

                sermon.title = recipes.getJSONObject(i).getString("title");
                sermon.description = recipes.getJSONObject(i).getString("description");

                sermonList.add(sermon);
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }

        return sermonList;
    }

    //test
    private static String loadJsonFromAsset(String filename) {
        String json = null;

        try {
            Context context = MyApplication.getAppContext();
            InputStream is = context.getAssets().open(filename);
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();
            json = new String(buffer, "UTF-8");
        }
        catch (java.io.IOException ex) {
            ex.printStackTrace();
            return null;
        }

        return json;
    }

}
